File: 4_Sorting_Algorithms/Insertion_Sort/C++/insertion_sort.cpp
#include "insertion_sort.hpp"

/**
 * insertion_sort - Sorts an array using the insertion sort algorithm.
 *
 * @arr: Reference to the array to be sorted.
 */
void insertion_sort(std::vector<int> &arr) {
    int len_vector = static_cast<int>(arr.size());

    for (int i = 1; i < len_vector; i++) {
        int key = arr[i];
        int j = i - 1;

        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
}

File: 4_Sorting_Algorithms/Insertion_Sort/Python/insertion_sort.py
from typing import List


def insertion_sort(arr: List[int]):
    """
    Sorts an array using the insertion sort algorithm.

    :param arr: List of integers to be sorted.
    """
    for i in range(1, len(arr)):
        key = arr[i]
        j = i - 1

        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j -= 1
        
        arr[j + 1] = key
